pkgs/applications/misc/hugo/default.nix

```
@@ -33,7 +33,7 @@ buildGoModule rec {
   '';
 
   meta = with lib; {
-    broken = stdenv.isDarwin;
+    broken = stdenv.isDarwin && stdenv.isx86_64;
```

It will still be nice to document why this is broken :smile_cat: 

```suggestion
    # TODO: Remove once nixpkgs uses macOS SDK 10.14+ for x86_64-darwin
    # Undefined symbols for architecture x86_64: "_SecTrustEvaluateWithError"
    # https://github.com/NixOS/nixpkgs/issues/101229
    broken = stdenv.isDarwin && stdenv.isx86_64;
```

---

@bobby285271 I mean, for a "proper" fix, can we test the SDK since that seems to be what's specifically the issue?
something like
```
builtins.compareVersions darwin.apple_sdk.MacOSX-SDK.version "10.13" < 0
```


Other packages also seem to do something like
```
++ lib.optionals stdenv.isDarwin [ CoreFoundation Security ];
```
Perhaps these could be combined and applied globally to patch over things until the SDK bump?

---

IMO we don't need that comment. Half of the darwin packages are broken because of that and if that is fixed it probably needs manual fixes, anyway.

---

turns out the darwin sdk version is already in stdenv;
```
broken = (stdenv.isDarwin && builtins.compareVersions stdenv.hostPlatform.darwinSdkVersion "10.13" < 0);
```

@SuperSandro2000 does this look more reasonable?

---

one possible timeline/plan of action:

1. present day: macos sdk when `stdenv.isx86_64` is a museum piece. contemporary software is being erroneously marked wholesale broken on darwin as a result

2. `stdenv` gains a member:
   ```
   isDarwinSDKStillAncient =
     (isDarwin && lib.versionOlder hostPlatform.darwinSdkVersion "10.13")
   ```
   (name is a working title and up for debate)

3. packages gradually replace `broken = isDarwin` with `broken = isDarwinSDKStillAncient`

4. (optional) PRs newly marking packages as broken on darwin are specially tagged and a github actions bot called clippy.exe autoreplies "did you mean `isDarwinSDKStillAncient`?"
   * this way, well-intentioned committers chasing zhf who nonetheless do not actually use nix on macos don't accidentally break things

5. some indeterminate span of time passes

6. through herculean effort, the macos sdk ceases to be ancient on x86_64-darwin

7. a number of packages are suddenly no longer marked as broken; a majority of build just fine, while the minority of build failures are attended to on a case-by-case basis

8. nixpkgs is scoured for references to `isDarwinSDKStillAncient`; definition and all references are removed

9. by this time, apple have probably stopped shipping macos for x86_64 anyway

10. we reflect fondly on the fact that between (3) and (6), we enjoyed a relative lack of spurious breakages on aarch64-darwin

---

Didn't read the checklist. My suggestion is to merge as is because we don't do it different for other packages.